detox/runners/jest-circus/utils/assertJestCircus27.js

```diff
@@ -0,0 +1,56 @@
+const fs = require('fs');
+const path = require('path');
+
+const semver = require('semver');
+
+const detoxPackageJson = require('../../../package.json');
+const DetoxRuntimeError = require('../../../src/errors/DetoxRuntimeError');
+
+function assertJestCircus27(maybeProjectConfig) {
+  const projectConfig = maybeProjectConfig.projectConfig || maybeProjectConfig;
+
+  if (!/jest-circus/.test(projectConfig.testRunner)) {
+    throw new DetoxRuntimeError({
+      message: 'Cannot continue without Jest Circus test runner underneath, exiting.',
+      hint: 'Make sure that in your Jest config you have no "testRunner" property or it is explicitly set to "jest-circus/runner".',
+      debugInfo: `The test runner powering your configuration is:\n${projectConfig.testRunner}`,
+    });
+  }
+
+  const circusPackageJson = path.join(path.dirname(projectConfig.testRunner), 'package.json');
+  if (!fs.existsSync(circusPackageJson)) {
+    throw new DetoxRuntimeError({
+      message: 'Check that you have an installed copy of "jest-circus" npm package, exiting.',
+      debugInfo: `Its package.json file is missing: ${circusPackageJson}`,
+    });
+  }
+
+  const circusVersion = require(circusPackageJson).version;
+  if (!circusVersion) {
+    throw new DetoxRuntimeError({
+      message: 'Check that you have an valid copy of "jest-circus" npm package, exiting.',
+      debugInfo: `Its package.json file has no "version" property. See:\n` + circusPackageJson,
+    });
+  }
+
+  assertSupportedVersion(circusVersion);
+
+  return maybeProjectConfig;
+}
+
+function assertSupportedVersion(actualVersion) {
```

Comment:
On a nit-picky note related strictly to styling: Why have a specific function just for that last part (version assertion), and not for any of the preliminary assertions as well? 🤔